pgctl/cli.py

```diff
@@ -289,14 +322,41 @@ def idempotent_supervise(self):
                 ('supervise', service.strpath),
                 stdout=service.join('stdout.log').open('w'),
                 stderr=service.join('stderr.log').open('w'),
+                env=self.supervise_env(service),
             )  # pragma: no branch
             # (see https://bitbucket.org/ned/coveragepy/issues/146)
 
+    def supervise_env(self, service_path):
```

Comment:
done. I spent a lot of time going back and forth on how to scope the class, I think where it is now is reasonable, but open to suggestions

I moved most of the service-specific stuff to a separate service module. The cli now always works in Service objects to avoid confusion (is `service` a LocalPath? a path string? a Service?).

A separate module seems reasonable, potentially we might want to use these things separately from the cli. I'd also like to move the daemontools stuff out to a daemontools module (so that `cli` only deals with cli-related stuff), but in a separate review.